### PR TITLE
Comments API - pure readr server comment API

### PR DESCRIPTION
**Pull request by FalseChord**
- Comments API
  Comment API routes/models/tests/configs
- DB modification
  Add comments, comments_reported tables
  Set Member.mail to Not Null

### CODE REVIEW

File: db_schema/readrsql.patch.6.up.sql [hunk 1]
```diff
@@ -0,0 +1,6 @@
+CREATE TABLE comments ( id BIGINT UNSIGNED PRIMARY KEY AUTO_INCREMENT, body varchar(500), og_title varchar(256), og_description varchar(256), og_image varchar(128), like_amount int, parent_id bigint UNSIGNED, resource varchar(128), created_at datetime, updated_at datetime, author BIGINT UNSIGNED, active tinyint, status tinyint, ip int unsigned);
```
Comment (hcchien) [inline db_schema/readrsql.patch.6.up.sql]: ip int unsigned?

File: db_schema/readrsql.patch.6.up.sql [hunk 2]
```diff
@@ -0,0 +1,6 @@
...
+
+CREATE TABLE comments_reported ( id BIGINT UNSIGNED PRIMARY KEY AUTO_INCREMENT, comment_id BIGINT UNSIGNED, reporter BIGINT UNSIGNED, reason varchar(500), solved tinyint, created_at datetime, updated_at datetime);
```
Comment (hcchien) [inline db_schema/readrsql.patch.6.up.sql]: need ip here.

File: db_schema/readrsql.patch.6.up.sql (hunk 1 above)
Comment (FalseChord) [reply db_schema/readrsql.patch.6.up.sql]: According to this post: 
https://stackoverflow.com/questions/5133580/which-mysql-datatype-to-use-for-an-ip-address?utm_medium=organic&utm_source=google_rich_qa&utm_campaign=google_rich_qa

Store it as unsigned int is an option.
Or should I use string datatype to store IP value?

Comment (FalseChord) [reply db_schema/readrsql.patch.6.up.sql]: Just not sure which option is better.

Comment (hcchien) [reply db_schema/readrsql.patch.6.up.sql]: ok, unsigned int could be faster in data access.

File: db_schema/readrsql.patch.6.up.sql (hunk 2 above)
Comment (FalseChord) [reply db_schema/readrsql.patch.6.up.sql]: Do we also need IP column for comments_reported table?

Comment (hcchien) [reply db_schema/readrsql.patch.6.up.sql]: yes.

**Pull request merged**